js/index.js

```diff
@@ -1,23 +1,40 @@
 // ITERATION 1
 
 function updateSubtotal(product) {
-  console.log('Calculating subtotal, yey!');
+  //console.log('Calculating subtotal, yey!');
 
   //... your code goes here
+  const price = product.querySelector('.price span').innerText;
+  const quantity = product.querySelector('.quantity input').value;
+  const subtotal = price * quantity;
+  product.querySelector('.subtotal span').innerText = subtotal.toFixed(2);
+  //console.log('updateSubtotal', price, quantity, subtotal);
 }
 
 function calculateAll() {
   // code in the following two lines is added just for testing purposes.
   // it runs when only iteration 1 is completed. at later point, it can be removed.
-  const singleProduct = document.querySelector('.product');
-  updateSubtotal(singleProduct);
+  //const singleProduct = document.querySelector('.product');
+  //updateSubtotal(singleProduct);
   // end of test
 
   // ITERATION 2
   //... your code goes here
+  const products = document.querySelectorAll('.product');
+  products.forEach(function (singleProduct) {
+    updateSubtotal(singleProduct);
+  });
 
   // ITERATION 3
   //... your code goes here
+  const totalPricesDom = document.querySelectorAll('.product .subtotal span');
+  let totalPrice = 0;
+  for (let domEl of totalPricesDom) {
+    totalPrice += Number.parseFloat(domEl.innerText);
+  }
+  document.querySelector('#total-value span').innerText = totalPrice.toFixed(2);
+  //console.log(document.querySelector('.product .subtotal span').innerText);
+  // console.log('totalPrice', totalPrice);
```

Comment:
Good Job! Another way to do it is to make `updateSubtotal` function return the subtotal and then you just accumulate it in the loop above in line 25.